core/src/test/java/mediumtest/SmartNotificationsMediumTests.java

```diff
@@ -0,0 +1,125 @@
+  private static final String PROJECT_KEY = "projectKey";
+  private static final String PROJECT_KEY_2 = "projectKey2";
+  private static final String CONNECTION_ID = "connectionId";
+  private static final String CONNECTION_ID_2 = "connectionId2";
+  private static final String DATETIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ssZ";
+  private static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern(DATETIME_FORMAT);
+  private static final String EVENT = "{\"events\": [" +
+    "{" +
+    "\"message\": \"msg\"," +
+    "\"link\": \"lnk\"," +
+    "\"project\": \"projectKey\"," +
+    "\"date\": \"2022-01-01T08:00:00+0000\"," +
+    "\"category\": \"category\"" +
+    "}" +
+    "]" +
+    "}";
+  private static final String TWO_EVENTS = "{\"events\": [" +
+    "{" +
+    "\"message\": \"msg\"," +
+    "\"link\": \"lnk\"," +
+    "\"project\": \"projectKey\"," +
+    "\"date\": \"2022-01-01T08:00:00+0000\"," +
+    "\"category\": \"category\"" +
+    "},{" +
+    "\"message\": \"msg\"," +
+    "\"link\": \"lnk\"," +
+    "\"project\": \"projectKey\"," +
+    "\"date\": \"2022-01-01T08:00:00+0000\"," +
+    "\"category\": \"category\"" +
+    "}]" +
+    "}";
+
+  @AfterEach
+  void tearDown() throws ExecutionException, InterruptedException {
+    backend.shutdown().get();
+  }
+
+  @Test
+  void it_should_send_notification(@TempDir Path tmpDir) {
+    var fakeClient = newFakeClient().build();
+    
+    writeFile(tmpDir, CONNECTION_ID, PROJECT_KEY);
+    writeFile(tmpDir, CONNECTION_ID_2, PROJECT_KEY);
+    writeFile(tmpDir, CONNECTION_ID_2, PROJECT_KEY_2);
+
+    mockWebServerExtension.addResponse("/api/developers/search_events?projects=&from=", new MockResponse().setResponseCode(200));
+    mockWebServerExtension.addStringResponse("/api/developers/search_events?projects=" + PROJECT_KEY + "&from=" +
+      UrlUtils.urlEncode(STORED_DATE.format(TIME_FORMATTER)), EVENT);
+    mockWebServerExtension.addStringResponse("/api/developers/search_events?projects=" + PROJECT_KEY + "," + PROJECT_KEY_2 + "&from=" +
+      UrlUtils.urlEncode(STORED_DATE.format(TIME_FORMATTER)) + "," + UrlUtils.urlEncode(STORED_DATE.format(TIME_FORMATTER)), TWO_EVENTS);
+
+    backend = newBackend()
+      .withSonarQubeConnection(CONNECTION_ID, mockWebServerExtension.endpointParams().getBaseUrl())
+      .withSonarQubeConnection(CONNECTION_ID_2, mockWebServerExtension.endpointParams().getBaseUrl())
+      .withBoundConfigScope("scopeId", CONNECTION_ID, PROJECT_KEY)
+      .withBoundConfigScope("scopeId2", CONNECTION_ID, PROJECT_KEY)
+      .withBoundConfigScope("scopeId3", CONNECTION_ID_2, PROJECT_KEY)
+      .withBoundConfigScope("scopeId4", CONNECTION_ID_2, PROJECT_KEY_2)
+      .withStorageRoot(tmpDir)
+      .build(fakeClient);
+
+    await().atMost(3, TimeUnit.SECONDS).until(() -> !fakeClient.getSmartNotificationsToShow().isEmpty());
+
+    assertThat(fakeClient.getSmartNotificationsToShow()).hasSize(3);
```

Comment:
If possible I would assert the content of the notifications

Reply:
Yes! Tests in progress :keyboard: